### PR TITLE
Add showDefaultInputIcon and customInputIcon for SingleDatePicker

### PR DESCRIPTION
**Pull request by b0gok**
According to issue https://github.com/airbnb/react-dates/issues/368

### CODE REVIEW

**Comment by majapw** [comment]
Awesome. I will take a look at this tomorrow!

**Comment by JemarJones** [comment]
This would be super useful for me to have right now. @majapw How goes the reviewal?

**Comment by ljharb** [comment]
@b0gok please rebase on the command line; the "update" button on github creates a merge commit.

File: src/components/SingleDatePicker.jsx [hunk 1]
```diff
@@ -54,6 +54,8 @@ const defaultProps = {
   readOnly: false,
   screenReaderInputMessage: '',
   showClearDate: false,
+  showDefaultInputIcon: false,
+  customInputIcon: null,
```
Comment (majapw) [inline src/components/SingleDatePicker.jsx]: I think we need to pass this through as well, no?

File: examples/SingleDatePickerWrapper.jsx [hunk 2]
```diff
@@ -36,6 +36,9 @@ const defaultProps = {
   required: false,
   screenReaderInputMessage: '',
   showClearDate: false,
+  showDefaultInputIcon: false,
+  customInputIcon: null,
+  customCloseIcon: null,
```
Comment (majapw) [inline examples/SingleDatePickerWrapper.jsx]: This change isn't adding a `customCloseIcon` is it?

File: stories/SingleDatePicker_input.js [hunk 3]
```diff
@@ -41,6 +54,18 @@ storiesOf('SDP - Input Props', module)
       displayFormat="MMM D"
     />
   ))
+  .addWithInfo('with show calendar icon', () => (
+    <SingleDatePickerWrapper
+      initialDate={moment().add(3, 'days')}
+      showDefaultInputIcon
+    />
+  ))
+  .addWithInfo('with custom show calendar icon', () => (
```
Comment (majapw) [inline stories/SingleDatePicker_input.js]: Does this work right now? We don't appear to be passing through the `customInputIcon`

File: src/components/SingleDatePicker.jsx (hunk 1 above)
Comment (b0gok) [inline src/components/SingleDatePicker.jsx]: Yes, you are right, my mistake 🙁

File: examples/SingleDatePickerWrapper.jsx (hunk 2 above)
Comment (b0gok) [inline examples/SingleDatePickerWrapper.jsx]: Removed it

File: stories/SingleDatePicker_input.js (hunk 3 above)
Comment (b0gok) [inline stories/SingleDatePicker_input.js]: Now it should work properly

**Comment by b0gok** [comment]
> Don't forget to add them to the SingleDatePickerShape file! :) I think that's the last thing.

Done (:

**Comment by JemarJones** [comment]
I would be forever grateful if this release happened soon 🤞

**Comment by majapw** [comment]
v12.1.0 has been released!

**Comment by lopezdp** [comment]
Are there any examples for this use case documented?

**Comment by b0gok** [comment]
> Are there any examples for this use case documented?

@lopezdp [Here](https://airbnb.io/react-dates/?path=/story/sdp-input-props--with-show-calendar-icon)  and [here](https://airbnb.io/react-dates/?path=/story/sdp-input-props--with-custom-show-calendar-icon)

**Pull request merged**